pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -204,6 +218,43 @@ func (pl *CSILimits) filterAttachableVolumes(
 	return nil
 }
 
+// checkAttachableInlineVolume takes an inline volume and add to the result map if the
+// volume is migratable and CSI migration for this plugin has been enabled.
+func (pl *CSILimits) checkAttachableInlineVolume(vol v1.Volume, csiNode *storagev1.CSINode,
+	pod *v1.Pod, result map[string]string) error {
+	if !pl.translator.IsInlineMigratable(&vol) {
+		return nil
+	}
+	// Check if the intree provisioner CSI migration has been enabled.
+	inTreeProvisionerName, err := pl.translator.GetInTreePluginNameFromSpec(nil, &vol)
+	if err != nil {
+		return fmt.Errorf("looking up provisioner name for volume %v: %w", vol, err)
+	}
+	if !isCSIMigrationOn(csiNode, inTreeProvisionerName) {
+		klog.V(5).InfoS("CSI Migration is not enabled for provisioner", "provisioner", inTreeProvisionerName,
+			"pod", klog.KObj(pod), "csiNode", csiNode.Name)
```

Comment:
Is it possible under regular circumstances that csiNode is nil?

Reply:
csiNode is checked in L96. And the Lister implementation [returns not found error](https://github.com/kubernetes/kubernetes/blob/9d01f0bf52891d777078daafcd146b6eea1c98eb/staging/src/k8s.io/client-go/listers/storage/v1/csinode.go#L65)

Reply:
It looks like the error is ignored here: https://github.com/kubernetes/kubernetes/blob/cc68c06f9cb6c70cbb840a8d7b9ca49eb223cb3f/pkg/scheduler/framework/plugins/nodevolumelimits/csi.go#L99

Reply:
Ah good catch. Yes this could be a problem. We do have logic that's supposed to block the node from being ready [until CSINode is created](https://github.com/kubernetes/kubernetes/blob/e4d46148de9d83c23001323d6613618c2897a427/pkg/volume/csi/csi_plugin.go#L299), but there could still be ordering issues with the informers.

Can you open a bug?

Reply:
It's referring to max supported 2-version skew between the control plane and kubelet. When CSI migration for a plugin is GA in version X, then in version X+2 we can assume that kubelets will be on version X. 

However, I think due to https://github.com/kubernetes/kubernetes/issues/115178, we realize now that autoscaler will encounter a nil CSINode always (because it doesn't simulate CSINode objects). So this comment is obsolete.